src/modules/accounting/extra_data.py

```diff
@@ -0,0 +1,153 @@
+        extra_data_bytes = self.to_bytes(extra_data)
+
+        data_format = FormatList.EXTRA_DATA_FORMAT_LIST_NON_EMPTY if extra_data else FormatList.EXTRA_DATA_FORMAT_LIST_EMPTY
+        return ExtraData(
+            extra_data=extra_data_bytes,
+            data_hash=Web3.keccak(extra_data_bytes),
+            format=data_format.value,
+            items_count=len(extra_data),
+        )
+
+    @staticmethod
+    def build_validators_payloads(
+        validators: dict[NodeOperatorGlobalIndex, int],
+        max_items_in_payload: int,
+        max_items_total: int,
+    ) -> tuple[list[ItemPayload], int]:
+        # sort by module id and node operator id
+        operator_validators = sorted(validators.items(), key=lambda x: x[0])
+        payload_size_limit = min(max_items_in_payload, max_items_total)
+
+        payloads = []
+
+        for module_id, operators_by_module in itertools.groupby(operator_validators, key=lambda x: x[0][0]):
+            operator_ids = []
+            vals_count = []
+
+            for ((_, no_id), validators_count) in operators_by_module:
+                operator_ids.append(no_id.to_bytes(ExtraDataService.Lengths.NODE_OPERATOR_IDS))
+                vals_count.append(validators_count.to_bytes(ExtraDataService.Lengths.STUCK_OR_EXITED_VALS_COUNT))
+                payload_size_limit -= 1
+
+                if not payload_size_limit:
+                    break
+
+            payloads.append(
+                ItemPayload(
+                    module_id=module_id.to_bytes(ExtraDataService.Lengths.MODULE_ID),
+                    node_ops_count=len(operator_ids).to_bytes(ExtraDataService.Lengths.NODE_OPS_COUNT),
+                    node_operator_ids=b"".join(operator_ids),
+                    vals_counts=b"".join(vals_count),
+                )
+            )
+
+            if not payload_size_limit:
+                break
+
+        return payloads, payload_size_limit
+
+    @staticmethod
+    def build_extra_data(stuck_payloads: list[ItemPayload], exited_payloads: list[ItemPayload]):
+        index = 0
+        extra_data = []
+
+        for item_type, payloads in [
+            (ItemType.EXTRA_DATA_TYPE_STUCK_VALIDATORS, stuck_payloads),
+            (ItemType.EXTRA_DATA_TYPE_EXITED_VALIDATORS, exited_payloads),
+        ]:
+            for payload in payloads:
+                extra_data.append(ExtraDataItem(
+                    item_index=index.to_bytes(ExtraDataService.Lengths.ITEM_INDEX),
+                    item_type=item_type,
+                    item_payload=payload
+                ))
+                index += 1
+
+        return extra_data
+
+    @staticmethod
+    def to_bytes(extra_data: list[ExtraDataItem]) -> bytes:
+        extra_data_bytes = b''
+        for item in extra_data:
+            extra_data_bytes += item.item_index
+            extra_data_bytes += item.item_type.value.to_bytes(ExtraDataService.Lengths.ITEM_TYPE)
```

Comment:
why `to_bytes` for only one parameter?

Reply:
it seems like pycharm doesn't know yet about python 3.11 changes
https://docs.python.org/3/library/stdtypes.html#int.to_bytes
`Changed in version 3.11: Added default argument values for length and byteorder.`

Reply:
We use item_type outside, so cant do encoding before